authentik/core/models.py

```diff
@@ -428,6 +429,16 @@ class Meta:
         abstract = True
 
 
+class ApplicationQuerySet(QuerySet):
+    def with_provider(self) -> "QuerySet[Application]":
+        qs = self.select_related("provider")
+        for subclass in Provider.objects.get_queryset()._get_subclasses_recurse(Provider):
```

Comment:
We could/should replace this with `Provider.__subclasses__()` but then it's annoying to convert the class into the lookup field.